File: src/ast/ArraySubscriptingExpression.java
package ast;

import java.io.StringWriter;

public class ArraySubscriptingExpression extends BaseNode {
    private BaseNode _leftNode;
    private BaseNode _subscript;

    public ArraySubscriptingExpression(BaseNode leftNode, BaseNode subscript) {
        super(NodeType.ArraySubscriptingExpression);
        _leftNode = leftNode;
        _subscript = subscript;
    }

    @Override
    public void PrintLeft(StringWriter writer) {
        writer.write("(");
        _leftNode.Print(writer);
        writer.write(")[");
        _subscript.Print(writer);
        writer.write("]");
    }
}

File: src/de/fearlesstobi/demangler/ast/ArraySubscriptingExpression.kt
package de.fearlesstobi.demangler.ast

import java.io.StringWriter

class ArraySubscriptingExpression(private val leftNode: BaseNode, private val subscript: BaseNode?) : BaseNode(NodeType.ArraySubscriptingExpression) {
    public override fun printLeft(writer: StringWriter) {
        writer.write("(")
        leftNode.print(writer)
        writer.write(")[")
        subscript!!.print(writer)
        writer.write("]")
    }

}